### ISSUE TITLE
Error Cloning repository using GitHub SCM Plugin

### ISSUE THREAD

**Issue by treezio** [updatecli/updatecli#1732]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

This scenario is kind of weird.
I'm currently setting up the updatecli pipeline for a "big" repository based on GitHub Actions, using the `GITHUB_TOKEN` as authentication.
Even though I have been able to clone many private repositories, I'm facing errors when attempting to clone this GitHub private repository using the `GitHub` SCM Plugin. 
I have recreated the errors in my local using an Ubuntu docker Image.

The main issue is that although I'm setting the `GITHUB_TOKEN` environment variable, it seems like it is trying to clone the repository using the SSH protocol, and it is complaining about not being able to create an SSH Agent.

```
+++++++++++
+ PREPARE +
+++++++++++

Loading Pipeline "./path/to/app/updatecli.yaml"

SCM repository retrieved: 1
ERROR: failed cloning GitHub repository "https://github.com/organization/repository.git"
ERROR: err - error creating SSH agent: "SSH agent requested but SSH_AUTH_SOCK not-specified"
```

I've tried to dive a bit in the `go-git` library but I have no clue why this is actually happening.

Maybe I'm missing something but I think this is not related to authentication errors, as I have been able to clone the repository using the `git` cli and the `GITHUB_TOKEN` via `https://` URL 

Pipeline Config:

```
---
name: Bump aplication chart version

scms:
  default:
    kind: github
    spec:
      user: "{{ .github.user }}"
      email: "{{ .github.email }}"
      owner: "{{ .github.owner }}"
      repository: "{{ .github.repository }}"
      token: "{{ requiredEnv .github.token }}"
      username: "{{ .github.username }}"
      branch: "{{ .github.branch }}"
```

Config values:
```
github:
  user: "SomeUser"
  email: "mail@organization.es"
  username: "treezio"
  token: "GITHUB_TOKEN"
  branch: "master"
  owner: "organization"
  repository: "repository"
```

I've tried this very same configuration just changing the `repository` value and it works.

### Expected Behavior

The repository is cloned via API using the authentication token `GITHUB_TOKEN`  

### Steps To Reproduce

- Run ubuntu 22.04 docker image in repository directory
- `docker run -ti -v $(pwd):/opt ubuntu:22.04`
- Install updatecli
- export `GITHUB_TOKEN` 
- Run `updatecli diff`

As this is happening with a private repository I haven't been able to recreate it with a public repository so far 😢 

### Environment

```markdown
- OS: ubuntu 22.04
- updatecli: 0.65.1
```

### Anything else?

As I said this is a private repository and kind of big, so I don't know if that might have something to do with this issue

**Comment by olblak** [comment]
Well it's going to be hard to debug but here some open question:

Does that repository contains git submodule?
-> Because by default Updatecli will try to clone each of them which may be an issue depending on how credentials are configured

Does that repository contains a git submodule using ssh protocol? 
-> Because I am aware of this issue https://github.com/updatecli/updatecli/issues/1098 which is rooted in the go-git library

If you are not using git submodule, are you using ssh or http protocol ?
-> Because you only need to specify a GitHub token if you use the http protocol, otherwise Updatecli should reuse your ssh agent

**Comment by treezio** [comment]
Thanks for all these tips!
I actually thought it was related to submodules and I tried to get rid of them to do some test but unfortunately did not work at first. I guess I did something wrong...

After reading your message I changed the submodules url reference to use `https` instead of `ssh` and it worked!

Thanks for your support! 👏🏻

**Comment by treezio** [comment]
For the record:

I have been experiencing a non-expected issue (at least not expected to me)

I created a new branch in my gitops repository, the one I could not clone beforehand, I updated the references for the submodules to be fetched using https protocol and pushed the changes.

Then I updated my github.branch configuration to use the branch I just pushed instead of my `master` branch.

This did not work, I still had issues in the clone repository step. 

It worked after merging the changes to master.

I would expect it to work using the fixed branch I pushed to the remote repository.

The configuration is the same as the one I specified in the original message for this issue just updating the `github.branch` value
